pkg/operator/operator.go

```diff
@@ -401,6 +412,39 @@ func New(
 	)
 	o.informerFactories = append(o.informerFactories, kubeInformersOperatorNS)
 
+	configInformers := configv1informers.NewSharedInformerFactory(configClient, 10*time.Minute)
+	missingVersion := "0.0.1-snapshot"
+	desiredVersion := missingVersion
+	if envVersion, exists := os.LookupEnv("RELEASE_VERSION"); exists {
+		desiredVersion = envVersion
+	}
+
+	// By default, this will exit(0) if the featuregates change
+	featureGateAccessor := featuregates.NewFeatureGateAccess(
+		desiredVersion, missingVersion,
+		configInformers.Config().V1().ClusterVersions(),
+		configInformers.Config().V1().FeatureGates(),
+		eventRecorder,
+	)
+	go featureGateAccessor.Run(ctx)
+	go configInformers.Start(ctx.Done())
```

Comment:
I'm wondering if we shouldn't run the informer just like the others (by appending to `o.informerFactories`). Any reason not to do that?

Similarly for the `featureGateAccessor`: afaiu assigning o.metricsServerEnabled`` only runs once now `operator.New()`. Doesn't this mean that we can't detect a later change in the featuregates? Or is that intentional? I don't fully understand how `featuregates` work yet, so I might be missing something.

Reply:
Oh I see now, the accessor restarts the operator when feature gates change. In that case I'd kindly ask to extend the comment in line 422 a bit :)

Reply:
ya thats my understanding too. I updated the comment :)